src/components/BulkOperations/index.jsx

```diff
@@ -0,0 +1,194 @@
+import  { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import Papa from 'papaparse';
+
+import { SelectMenu, MenuItem, Dropzone, DataTable, Alert} from '@openedx/paragon';
+import Collapsible from '../Collapsible'
+import LoadingSpinner from '../LoadingSpinner';
+import { formatDate } from '../../utils';
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+
+import './BulkOperations.scss';
+
+const availableOperations = {
+    "course_create": "Bulk Create",
+    "course_partial_update": "Bulk Course Update",
+    "course_run_partial_update": "Bulk CourseRun Update",
+    "course_rerun": "Bulk Rerun"
+}
+
+
+function BulkOperations(){
+    const [bulkOperationId, setBulkOperationId] = useState(null);
+    const [fileContent, setFileContent] = useState(null);
+    const [file, setFile] = useState(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [submitSuccess, setSubmitSuccess] = useState(null);
+    const [historicalRecords, setHistoricalRecords] = useState(null);
+
+    function filteredHistoricalRecords() {
+        if (historicalRecords == null){
+            return []
+        }
+        return historicalRecords.filter(rec => !bulkOperationId || rec.task_type === bulkOperationId)
+    }
+
+    function handleUploadNew(){
+        setFileContent(null);
+        setFile(null);
+    }
+
+    function handleFileUpload({fileData}){
+        let file = fileData.get('file');
+        file.text().then(content => {setFile(file); setFileContent(content);});
+    }
+
+    async function handleSubmit(){
+        setIsLoading(true);
+        try {
+            const res = await DiscoveryDataApiService.createBulkOperation(file, bulkOperationId);
+            setSubmitSuccess(true);
+            setHistoricalRecords([res.data,...historicalRecords])
+        }
+        catch (e) {
+            setSubmitSuccess(false);
+        }
+        setIsLoading(false);
+    }
+
+    async function fetchBulkOpTasks(){
+        const response = await DiscoveryDataApiService.fetchBulkOperations();
+        setHistoricalRecords(response.data.results);
+    }
+
+    function getDropZone(){
+        return <Dropzone
+            onProcessUpload={handleFileUpload}
+        />
+    }
+
+    function getCollapsibleTitle() {
+        return <span className='font-weight-bold'>Processing History</span>
+    }
+
+    function getAlert() {
+        let variant, message;
+        if (submitSuccess == null) {
+            return null;
+        }
+
+        if (submitSuccess == true) {
+            variant = "success"
+            message = "Successfully submitted task for bulk operation"
+        }
+        else if (submitSuccess == false) {
+            variant = "danger"
+            message = "Failed to submit task for processing"
+        }
+
+        return <Alert variant={variant} dismissible onClick={() => {setSubmitSuccess(null)}}>{message}</Alert>
+    }
```

Comment:
Can we replace ifs with mapping structure?

```js
    const alertMapping = {
        true: {
            variant: "success",
            message: "Successfully submitted task for bulk operation",
        },
        false: {
            variant: "danger",
            message: "Failed to submit task for processing",
        },
    };
```

Reply:
There are really only two cases. If there were more, a mapping might help, but as is, I find the if-else more readable.